src/MSBuild/XMake.cs

```diff
@@ -1393,7 +1393,7 @@ .. distributedLoggerRecords.Select(d => d.CentralLogger)
 
                 projectCollection = new ProjectCollection(
                     globalProperties,
-                    evaluationLoggers,
+                    isPreprocess ? null : evaluationLoggers,
```

Comment:
this deserves a comment, why the condition is there

Reply:
@GangWang01 please handle it so we can proceed